src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -29,16 +29,24 @@ public class TelemetryActivity : IDisposable
         public Guid OperationId { get; }
 
         /// <summary> Singleton of NuGet telemetry service instance. </summary>
-        public static INuGetTelemetryService NuGetTelemetryService { get; set; }
+        public static INuGetTelemetryService? NuGetTelemetryService { get; set; }
 
         private TelemetryActivity(Guid parentId, TelemetryEvent telemetryEvent, Guid operationId)
         {
             if (telemetryEvent != null)
             {
+                if (telemetryEvent.Name is null)
+                {
+                    throw new ArgumentException(paramName: nameof(telemetryEvent), message: "Property 'Name' must not be null");
+                }
                 _telemetryActivity = NuGetTelemetryService?.StartActivity(telemetryEvent.Name);
```

Comment:
This exposes an API design problem. We have code creating instances of `TelemetryEvent` with a null name, but this code will throw a NullReferenceException if such an instance is used here.